pandoc-slave/pandoc-workflow/pandocrunner.cpp

```diff
@@ -0,0 +1,89 @@
+#include "pandocrunner.h"
+
+#include <QStringList>
+#include <QCoreApplication>
+#include <QDir>
+
+
+PandocRunner::PandocRunner(QString &pandocExePath, QObject* parent):
+    QProcess(parent)
+  , mContent("")
+  , mError("")
+  , mStatusCode(0)
+  , mPandocExePath(pandocExePath)
+{
+    mMetaEnum = QMetaEnum::fromType<PandocFormat>();
+
+    initializeConnections();
+}
+
+QString PandocRunner::run(PandocFormat from, PandocFormat to, const QString &file)
+{
+
+    QStringList params = buildParams(from, to, file);
+
+    start(mPandocExePath, params);
+
+    waitForFinished(-1);
```

Comment:
We could not use "waitForFinished", as it will block main thread that cause UI freeze. Please, rewrite it in asyncronious style: method start() will run process, then signal will notify about process finish.